llvm/test/CodeGen/AMDGPU/amdgpu-attributor-accesslist-offsetbins-out-of-sync.ll

```diff
@@ -0,0 +1,95 @@
+; CHECK-NEXT:    [[SHFL:%.*]] = shufflevector <3 x float> [[SPLAT_SPLATINSERT_I]], <3 x float> zeroinitializer, <4 x i32> zeroinitializer
+; CHECK-NEXT:    [[TMP3:%.*]] = addrspacecast ptr [[NUM_CLOSURE_I26_I]] to ptr addrspace(5)
+; CHECK-NEXT:    [[LOAD:%.*]] = load i32, ptr addrspace(5) [[TMP3]], align 4
+; CHECK-NEXT:    [[IDXPROM_I27_I:%.*]] = sext i32 [[LOAD]] to i64
+; CHECK-NEXT:    [[ARRAYIDX_I28_I:%.*]] = getelementptr [64 x %struct.ShaderClosure], ptr [[CLOSURE_I25_I]], i64 0, i64 [[IDXPROM_I27_I]]
+; CHECK-NEXT:    [[TMP4:%.*]] = addrspacecast ptr [[ARRAYIDX_I28_I]] to ptr addrspace(5)
+; CHECK-NEXT:    store <4 x float> [[SHFL]], ptr addrspace(5) [[TMP4]], align 16
+; CHECK-NEXT:    [[INC_I30_I:%.*]] = or i32 [[LOAD]], 1
+; CHECK-NEXT:    [[TMP5:%.*]] = addrspacecast ptr [[NUM_CLOSURE_I26_I]] to ptr addrspace(5)
+; CHECK-NEXT:    store i32 [[INC_I30_I]], ptr addrspace(5) [[TMP5]], align 4
+; CHECK-NEXT:    br label %[[WHILE_COND]]
+;
+entry:
+  %closure.i25.i = getelementptr i8, ptr %kg, i64 336
+  %num_closure.i26.i = getelementptr i8, ptr %kg, i64 276
+  br label %while.cond
+
+while.cond:
+  %0 = load i32, ptr %kg, align 4
+  %idxprom.i = zext i32 %0 to i64
+  switch i32 0, label %sw.bb92 [
+  i32 1, label %sw.bb92
+  i32 0, label %subd_triangle_patch.exit.i.i35
+  ]
+
+subd_triangle_patch.exit.i.i35:
+  %arrayidx.i27.i = getelementptr float, ptr %kg, i64 %idxprom.i
+  store float 0.000000e+00, ptr %arrayidx.i27.i, align 4
+  br label %while.cond
+
+sw.bb92:
+  %insert = insertelement <3 x i32> zeroinitializer, i32 %0, i64 0
+  %splat.splatinsert.i = bitcast <3 x i32> %insert to <3 x float>
+  %shfl = shufflevector <3 x float> %splat.splatinsert.i, <3 x float> zeroinitializer, <4 x i32> zeroinitializer
+  %load = load i32, ptr %num_closure.i26.i, align 4
+  %idxprom.i27.i = sext i32 %load to i64
+  %arrayidx.i28.i = getelementptr [64 x %struct.ShaderClosure], ptr %closure.i25.i, i64 0, i64 %idxprom.i27.i
+  store <4 x float> %shfl, ptr %arrayidx.i28.i, align 16
+  %inc.i30.i = or i32 %load, 1
+  store i32 %inc.i30.i, ptr %num_closure.i26.i, align 4
+  br label %while.cond
+}
+
+define amdgpu_kernel void @kernel() #0 {
+; CHECK-LABEL: define amdgpu_kernel void @kernel(
+; CHECK-SAME: ) #[[ATTR1:[0-9]+]] {
+; CHECK-NEXT:  [[ENTRY:.*:]]
+; CHECK-NEXT:    [[SD_I1111111111:%.*]] = alloca [0 x [0 x [0 x [0 x [0 x [0 x [0 x [0 x [0 x %struct.ShaderData]]]]]]]]], i32 0, align 16, addrspace(5)
+; CHECK-NEXT:    [[KGLOBALS_ASCAST1:%.*]] = addrspacecast ptr addrspace(5) [[SD_I1111111111]] to ptr
+; CHECK-NEXT:    [[NUM_CLOSURE_I_I:%.*]] = getelementptr i8, ptr addrspace(5) [[SD_I1111111111]], i32 276
+; CHECK-NEXT:    store <2 x i32> zeroinitializer, ptr addrspace(5) [[NUM_CLOSURE_I_I]], align 4
+; CHECK-NEXT:    call fastcc void @foo(ptr [[KGLOBALS_ASCAST1]])
+; CHECK-NEXT:    ret void
+;
+entry:
+  %sd.i1111111111 = alloca [0 x [0 x [0 x [0 x [0 x [0 x [0 x [0 x [0 x %struct.ShaderData]]]]]]]]], i32 0, align 16, addrspace(5)
```

Comment:
should clean up this 0 sized allocation, there's an llvm-reduce bug open to stop doing this